src/clj/puppetlabs/http/client/async.clj

```diff
@@ -145,20 +209,26 @@
        are `:text` and `:stream`, which will return a `String` or an
        `InputStream`, respectively.  Defaults to `:stream`.
    * :query-params - used to set the query parameters of an http request"
```

Comment:
Sad as it is, this is pretty much the best documentation we have about the request options at the moment.  Probably worth adding something here about how to specify a metric id.